src/atlas/array/helpers/ArrayForEach.h

```diff
@@ -82,6 +88,15 @@ namespace helpers {
 
 namespace detail {
 
+template <typename>
+struct IsTupleImpl : std::false_type {};
+
+template <typename... Args>
+struct IsTupleImpl<std::tuple<Args...>> : std::true_type {};
+
+template <typename Tuple>
+using IsTuple = std::enable_if_t<IsTupleImpl<Tuple>::value>*;
+
```

Comment:
You can create a 
```c++
template <typename Tuple>
constexpr bool is_tuple() {
   return IsTupleImpl<Tuple>::value;
}
```
or
```c++
template <typename Tuple>
inline constexpr bool is_tuple_v = IsTupleImpl<Tuple>::value;
```
And then us it as
```c++
template <typename ArrayViewTuple, typename = std::enable_if_t<is_tuple<ArrayViewTuple>()>>
void myfunction(ArrayViewTuple&&, ......) 
```
or
```c++
template <typename ArrayViewTuple, typename = std::enable_if_t<is_tuple_v<ArrayViewTuple>>>
void myfunction(ArrayViewTuple&&, ......) 
```

Reply:
Done. Again, much nicer.